cpp/src/neighbors/ann_mg/ann_mg.cuh

```diff
@@ -741,3 +747,11 @@ ann_mg_index<cagra::index<T, IdxT>, T, IdxT> distribute_cagra(
 }
 
 }  // namespace cuvs::neighbors::mg::detail
+
+#else
+
+static_assert(false,
+              "FORBIDEN_MG_ALGORITHM_IMPORT\n\n"
+              "Please recompile the cuVS library with MG algorithms.\n");
```

Comment:
`#error "FORBIDEN_MG_ALGORITHM_IMPORT\n\n"
              "Please recompile the cuVS library with CMake option `BUILD_MG_ALGOS=ON.\n"

Reply:
Also this error based ifdef should be in `cpp/include`, not `cpp/src` as users will include the former

Reply:
`CUVS_BUILD_MG_ALGOS` is only defined during the build of `libcuvs.so` not during the build of user's target binary, right? Do we want CMake to edit the header file before installation? I think this can be done with a header template, and the `configure_file` command. But, is this worth it?

Reply:
I agree that if there's a safeguard it only makes sense to add it to the header file in the include directory.
However, in my mind there's two goals :
1) Preventing a cuVS developer from including something that wasn't compiled. The linking process should fail already, but the solution you propose allows the developer to get an instructive error message.
2) Preventing a cuVS user from including something that wasn't compiled. Don't know if we want to do this.

The problem is that with the current solution a cuVS user would have to define `CUVS_BUILD_MG_ALGOS` to be able to read the declarations for SNMG ANN.

Reply:
Why would you prefer a user be able to read the declarations? Declaration without definitions will just lead to errors down the road, because compilation will succeed for the user but linking will fail if they accidentally include the header.

Reply:
cuVS user doesn't have to define CUVS_BUILD_SNMG_ALGOS in their build system, they just have to turn it on in cuVS CMake. By setting CUVS_BUILD_SNMG_ALGOS to a PUBLIC definition, the user will automatically inherit it.

Reply:
Just noticed the existence of the `$CONDA_PREFIX/lib/cmake/cuvs/cuvs-targets.cmake` file. CMake should indeed be able to apply the compile definitions to the downstream projects thanks to it. Sorry, not very familiar with all of this.